Calls: Added recording to config settings (#6090)
mattermost/docs: source/configure/plugins-configuration-settings.rst

.. include:: ../_static/badges/selfhosted-only.rst
  :start-after: :nosearch:

**True**: Allow channel admins to enable or disable calls on specific channels. It also allows participants in DMs/GMs to enable or disable calls.

> **False**: Only system admins will be able to enable or disable calls on specific channels.

Reviewer: Not related to recording but as part of GA we removed this `Enable on specific channels` setting and renamed the `Enable on all channels` which is now `Test mode`, description is currently as follows:

>When test mode is enabled, only system admins are able to start calls in channels. This allows testing to confirm calls are working as expected.

This is an optional field. Changing this setting requires a plugin restart to take effect.

Enable call recordings (beta)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. include:: ../_static/badges/ent-selfhosted-only.rst
  :start-after: :nosearch:

> **True**: Allow call hosts to record meeting video and audio on the cloud. Recordings include the entire call window view along with participants' audio track and any shared screen video.

Reviewer 2: What's the default value for this config setting?

Author: Good question. I think it's **false**. @streamer45, can you confirm?

Reviewer: Default would be **false** since the feature requires extra setup in order to work.

> **True**: Allow call hosts to record meeting video and audio on the cloud. Recordings include the entire call window view along with participants' audio track and any shared screen video.

Reviewer: > Allow call hosts to record meeting video and audio on the cloud.

Maybe not a big deal but I could see a few customers wondering what we mean by "cloud" here. I'd like to stress that recordings are not sent anywhere other than the Mattermost instance where the plugin runs.

Author: Very good point, thank you!

> **True**: Allow call hosts to record meeting video and audio on the cloud. Recordings include the entire call window view along with participants' audio track and any shared screen video.

Author: ```suggestion
**True**: Allow call hosts to record meeting video and audio. Recordings include the entire call window view along with participants' audio track and any shared screen video. Recordings are stored in Mattermost.
```

**False**: Call recording functionality is not available to hosts.

Changing this setting requires a plugin restart to take effect.

Job service URL
~~~~~~~~~~~~~~~

.. include:: ../_static/badges/ent-selfhosted-only.rst
  :start-after: :nosearch:

> The URL to the instance where call recordings will be stored.

Reviewer 2: Is there a default for this config setting?

Author: I can't find one in the file. @streamer45, is there a default value for this?

Reviewer: There's no default as it's a required setting that should be provided by the system admin pointing to a running service.

> The URL to the instance where call recordings will be stored.

Reviewer: I think the description should be changed here to reflect what this is really about. The URL points to a running job service where all the processing related to recordings happens. The recordings files produced are ultimately stored in Mattermost, through the existing file attachment mechanism.

> The URL to the instance where call recordings will be stored.

Author: The URL to a running job service where all the processing related to recordings happens. The recorded files produced are ultimately stored in Mattermost, through the existing file attachment mechanism.

> The URL to the instance where call recordings will be stored.

Author: ```suggestion
The URL to a running job service where all the processing related to recordings happens. The recorded files produced are ultimately stored in Mattermost, through the existing file attachment mechanism.
```

> This is a required field.

Reviewer: Changing this setting requires a restart as well.

> The URL to a running job service where all the processing related to recordings happens. The recorded files produced are ultimately stored in Mattermost, through the existing file attachment mechanism.

Author: ```suggestion
The URL to a running job service where all the processing related to recordings happens. The recorded files produced are stored in Mattermost through the existing file attachment mechanism.
```

Changing this setting requires a plugin restart to take effect.

.. config:setting:: plugins-enableonspecificchannels
  :displayname: Enable on specific channels (Plugins - Calls)
  :systemconsole: Plugins > Calls
  :configjson: N/A
  :environment: N/A

>   :description: Manage who enable or disable calls on specific channels (deprecated from Mattermost v7.7)

Reviewer 2: ```suggestion
  :description: Manage who can enable or disable calls on specific channels (deprecated from Mattermost v7.7)
```

Enable on specific channels
~~~~~~~~~~~~~~~~~~~~~~~~~~~

> *This setting is not configurable from Mattermost v7.7; it is hidden and always true*

Reviewer 2: ```suggestion
*Admins can't configure this setting from Mattermost v7.7; it's hidden and always enabled*
```
